contracts/examples/guards/BaseGuard.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+import {IERC165} from "../../interfaces/IERC165.sol";
+import {BaseTransactionGuard, Guard} from "../../base/GuardManager.sol";
+import {BaseModuleGuard, IModuleGuard} from "../../base/ModuleManager.sol";
+
+/**
+ * @title BaseGuard - Inherits BaseTransactionGuard and BaseModuleGuard.
+ */
+abstract contract BaseGuard is BaseTransactionGuard, BaseModuleGuard {
+    /**
+     * @notice Checks if the contract supports an interface.
+     * @param interfaceId The interface identifier.
+     * @return True if the interfaceId matches the interfaceId of Guard/IModuleGuard/IERC165, false otherwise.
+     */
+    function supportsInterface(bytes4 interfaceId) external view virtual override(BaseTransactionGuard, BaseModuleGuard) returns (bool) {
+        return
+            interfaceId == type(Guard).interfaceId || // 0xe6d7a83a
+            interfaceId == type(IModuleGuard).interfaceId || // 0xe1ab3a1a
+            interfaceId == type(IERC165).interfaceId; // 0x01ffc9a7
+    }
+
+    /**
+     * @notice Called by the Safe contract after a transaction is executed.
+     *         This function definition is required here to avoid compilation errors as both Guard and IModuleGuard have identical function declaration.
```

Comment:
I'm not a huge fan of this comment, as this is "documented" by the compiler (if you remove it, you would get a compiler error).

Also, shouldn't this be `abstract` in order to force deriving contracts to implement this?

Reply:
On removing function definition getting the below error:

```
contracts/examples/guards/BaseGuard.sol:10:1: TypeError: Derived contract must override function "checkAfterExecution". Two or more base classes define function with same name and parameter types.
```

Else, we can create a `IBaseGuard` interface that inherits `ITransactionGuard` and `IModuleGuard` but each contract would then have to implement `supportsInterface` which would void the purpose of having `BaseGuard`

Reply:
I mean is it possible to tell the Solidity compiler that the contracts that inherit `BaseGuard` need to provide a single `checkAfterExecution` function? 

I know there is an `abstract` keyword in Solidity, but I don't know what incantation is needed for this particular "expression".

Reply:
I think should suffice: https://github.com/safe-global/safe-smart-account/pull/758/commits/4153512e7d59ab55d7d99bd80a989a54632c9095